sesman/chansrv/sound.c

```diff
@@ -436,6 +469,206 @@ sound_process_output_formats(struct stream *s, int size)
     return 0;
 }
 
+#if defined(XRDP_FDK_AAC)
+
+/*****************************************************************************/
+static int
+sound_wave_compress_fdk_aac(char *data, int data_bytes, int *format_index)
+{
+    int rv;
+    int cdata_bytes;
+    char *cdata;
+
+    AACENC_ERROR error;
+    int aot;
+    int sample_rate;
+    int mode;
+    int bitrate;
+    int afterburner;
+    int channel_order;
+    AACENC_InfoStruct info = { 0 };
+    AACENC_BufDesc in_buf = { 0 };
+    AACENC_BufDesc out_buf = { 0 };
+    AACENC_InArgs in_args = { 0 };
+    AACENC_OutArgs out_args = { 0 };
+    void *in_buffer[1];
+    int in_identifier[1];
+    int in_size[1];
+    int in_elem_size[1];
+    void *out_buffer[1];
+    int out_identifier[1];
+    int out_size[1];
+    int out_elem_size[1];
+
+    rv = data_bytes;
+
+    if (g_client_does_fdk_aac == 0)
+    {
+        return rv;
+    }
+
+    if (g_fdk_aac_encoder == 0)
+    {
+        /* init fdk aac encoder */
+        LOG(0, ("sound_wave_compress_fdk_aac: using fdk aac"));
+
+        error = aacEncOpen(&g_fdk_aac_encoder, 0, 2);
+        if (error != AACENC_OK)
+        {
+            LOG(0, ("sound_wave_compress_fdk_aac: aacEncOpen() failed"));
+            return rv;
+        }
+
+        aot = 2; /* MPEG-4 AAC Low Complexity. */
+        //aot = 129; /* MPEG-2 AAC Low Complexity. */
```

Comment:
This was left over from the effort to figure out why MSTSC does not decode.  I'll remove it.